runners/jet-experimental/src/main/java/org/apache/beam/runners/jet/processors/AbstractParDoP.java

```diff
@@ -247,7 +255,12 @@ private void processBufferedRegularItems(Inbox inbox) {
 
   @Override
   public boolean tryProcess() {
-    return outputManager.tryFlush();
+    boolean successful = outputManager.tryFlush();
+    if (successful && System.currentTimeMillis() > lastMetricsFlushTime + metricsFlushPeriod) {
```

Comment:
Should use `System.nanoTime`, real-time clock can move back and forth. Also you can use `nextMetricsFlushTime` and save one add operation for most calls.